### PR TITLE
feat(breadcrumbs): add isDarkBackground prop

### LINKED ISSUE #6724: Breadcrumbs component should have a color / backgroundcolor prop
### Description

Breadcrumbs component should have a color / backgroundcolor prop for the dark background screens

### Suggested solution

Additional prop for color/background color 

For example:
image

### Demo URL

_No response_

### Alternatives

_No response_

### Additional context

_No response_

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

### PR DESCRIPTION
**Pull request by nuria1110**
fix #6724

### Proposed behaviour

Add `isDarkBackground` prop to `Breadcrumbs` to allow consumers to change the colour scheme of the component to be on a darker background.

image

### Current behaviour

`Breadcrumbs` does not have a dark background variation. 

### Checklist

- [x] Commits follow our style guide
- [x] Related issues linked in commit messages if required
- [x] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Playwright automation tests added or updated if required
- [x] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [x] Related docs have been updated if required

#### QA

- [ ] Tested in provided StackBlitz sandbox/Storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### CODE REVIEW

File: src/components/breadcrumbs/crumb/crumb.test.tsx [hunk 1]
```diff
@@ -3,56 +3,51 @@ import { render, screen } from "@testing-library/react";
...
-        Link text
-      </Crumb>
-    );
-
-    const link = screen.getByText("Link text");
-    await userEvent.click(link);
-
-    expect(handleClickFn).toHaveBeenCalledTimes(0);
-  });
+test("passes href to the anchor element isCurrent is false", () => {
```
Comment (nineteen88) [inline src/components/breadcrumbs/crumb/crumb.test.tsx:6]: praise: nice one for tidying this up while you're in here!

**Review by nineteen88 — APPROVED** [review]

File: src/components/breadcrumbs/crumb/crumb.style.ts [hunk 2]
```diff
@@ -3,35 +3,50 @@ import Link, { LinkProps } from "../../link";
...
-  font: var(--typographyBreadcrumbSeparatorM);
-  color: var(--colorsUtilityYin055);
+interface DividerProps {
+  isDarkBackground: boolean;
+}
+
+export const Divider = styled.span<DividerProps>`
+  ::after {
+    content: "/";
+    margin: 0px var(--spacing050) 0px var(--spacing100);
```
Comment (DipperTheDan) [inline src/components/breadcrumbs/crumb/crumb.style.ts:40]: **nitpick/suggestion(non-blocking):** The design token for `0px` is `spacing000`. I'm happy if you leave it as is or want to change it to a token value.

File: src/components/breadcrumbs/crumb/crumb.style.ts [hunk 3]
```diff
@@ -3,35 +3,50 @@ import Link, { LinkProps } from "../../link";
...
+    line-height: 16px;
```
Comment (DipperTheDan) [inline src/components/breadcrumbs/crumb/crumb.style.ts:41]: **suggestion(non-blocking):** If you want to use a token value here, this is the equivalent for `16px`. I appreciate you've just moved this block of code and haven't written it from scratch, so I'm happy for you to leave it as is if you want 😄  

```suggestion
    line-height: var(--sizing200);
```

**Review by DipperTheDan — APPROVED** [review]

**Review by paulrobinson88 — APPROVED** [review]

**Pull request merged**